example/query_matching.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:pact_dart/pact_dart.dart';
+
+void main() {
+  // Create a Pact between the consumer and provider
+  final pact = PactMockService('QueryMatcherConsumer', 'QueryMatcherProvider');
+
+  // Create a new interaction
+  pact
+      .newInteraction(description: 'a request to search with query matchers')
+      // Set up the provider state
+      .given('search service is running')
+      // Configure the request
+      .withRequest('GET', '/api/search', query: {
+    // Simple value
+    'simple': 'value',
+    // Multiple values for a single parameter
+    'ids': PactMatchers.QueryMultiValue(['1', '2', '3']),
+    // Regex matching for a single value
+    'code': PactMatchers.QueryRegex('ABC123', '^[A-Z]{3}\\d{3}\$'),
```

Comment:
Just to confirm my understanding, is `QueryRegex` a developer shortcut to `Term` to help make the interaction more understandable?